app/controllers/carts_controller.rb

```diff
@@ -30,8 +33,35 @@ class CartsController < ApplicationController
       The `cart_items` doesn't support partial updates yet though and must be
       given in full for each update of `cart_items`.
     EON
-    param :path, :id, :integer, :required, 'The cart ID'
-    param :body, :cart, :string, :required, 'JSON representation of a cart'
+    param :path, :id, :integer, :required, 'Cart ID'
+    param :body, :cart, :writeCart, :required, 'Cart'
+    response :ok, 'Success', :readCart
+    response :bad_request
+    response :not_found
+  end
+
+  swagger_model :readCart do
+    property :id, :integer, :optional, 'Cart ID'
+    property :user_id, :integer, :optional, 'User ID'
+    property :cart_items, :array, :optional, 'Cart Items', {'items' => {'$ref' => 'readCartItem'}}
```

Comment:
Line is too long. [98/80]<br>Redundant curly braces around a hash parameter.<br>Space inside { missing.<br>Space inside } missing.